cfme/containers/replicator.py

```diff
@@ -42,6 +45,19 @@ def get_detail(self, *ident):
         self.load_details(refresh=True)
         return details_page.infoblock.text(*ident)
 
+    @classmethod
+    def get_random_instances(cls, provider, count=1, appliance=None):
```

Comment:
**REQUIRED-FOLLOWUP** I'm afraid I don't see it as redundant at all. Assuming you can't have a singluar prototype for this method, you can at least have a single prorotype for most of them. I count exactly 1 that is different. 9 of them are the same prototype. We can drop the appliance argument, as the appliance can be obtained from the provider object. ```provider.appliance``` to instantiate:

```python
    cls(chosen.name, chosen.project_name, provider, appliance=provider.appliance)
```

The ```itertools.islice``` as ronny mentioned is perfect for this and yields cleaner code:

```python
random.shuffle(rc_list)
return [cls(obj.name, obj.project_name, provider) for obj in itertools.islice(rc_list, count)]
```

This will remove the need for popping from the list, drop the while loop, making it overall easier to read and faster, (though I know it will only be a small number of elements).